osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/mapping/DataObjectToEventListConverter.java

```diff
@@ -123,36 +140,99 @@ private Integer extractEventCounter(
+      final Protocol protocol)
       throws ProtocolAdapterException {
 
+    if (eventLogCategory.getDetailsType() == EventLogDetailsType.MAGNITUDE) {
+      return this.extractMagnitudeThd(eventData, eventCode, protocol);
+    }
     if (eventLogCategory.getDetailsType() == EventLogDetailsType.MAGNITUDE_AND_DURATION) {
-      return this.extractMagnitudeAndDuration(eventData);
+      return this.extractMagnitudeAndDuration(eventData, eventCode, protocol);
     }
 
     return Collections.emptyList();
   }
 
-  private List<EventDetailDto> extractMagnitudeAndDuration(final List<DataObject> eventData)
+  private List<EventDetailDto> extractMagnitudeThd(
+      final List<DataObject> eventData, final Integer eventCode, final Protocol protocol)
       throws ProtocolAdapterException {
 
     final List<EventDetailDto> eventDetails = new ArrayList<>();
 
-    this.checkIsNumber(eventData.get(2), MAGNITUDE);
-    eventDetails.add(
-        new EventDetailDto(
-            MAGNITUDE,
-            this.readValueWithScalerAndUnit(eventData.get(2), BigDecimal.valueOf(0.1), "V")));
+    try {
+      final String scalerUnitMagnitude =
+          this.getScalerUnit(eventCode, protocol, DlmsObjectType.POWER_QUALITY_THD_EVENT_MAGNITUDE);
+
+      this.checkIsNumber(eventData.get(2), MAGNITUDE);
+      eventDetails.add(
+          new EventDetailDto(
+              MAGNITUDE, this.readValueWithScalerAndUnit(eventData.get(2), scalerUnitMagnitude)));
+    } catch (final ObjectConfigException e) {
+      throw new ProtocolAdapterException("Error in object config", e);
+    }
+    return eventDetails;
+  }
+
+  private List<EventDetailDto> extractMagnitudeAndDuration(
+      final List<DataObject> eventData, final Integer eventCode, final Protocol protocol)
+      throws ProtocolAdapterException {
+
+    final List<EventDetailDto> eventDetails = new ArrayList<>();
 
-    this.checkIsNumber(eventData.get(3), DURATION);
-    eventDetails.add(
-        new EventDetailDto(
-            DURATION,
-            this.readValueWithScalerAndUnit(eventData.get(3), BigDecimal.valueOf(0.1), "s")));
+    try {
+      final String scalerUnitMagnitude =
+          this.getScalerUnit(
+              eventCode, protocol, DlmsObjectType.POWER_QUALITY_EXTENDED_EVENT_MAGNITUDE);
+
+      this.checkIsNumber(eventData.get(2), MAGNITUDE);
+      eventDetails.add(
+          new EventDetailDto(
+              MAGNITUDE, this.readValueWithScalerAndUnit(eventData.get(2), scalerUnitMagnitude)));
+
+      final String scalerUnitDuration =
+          this.getScalerUnit(
+              eventCode, protocol, DlmsObjectType.POWER_QUALITY_EXTENDED_EVENT_DURATION);
+
+      this.checkIsNumber(eventData.get(3), DURATION);
+      eventDetails.add(
+          new EventDetailDto(
+              DURATION, this.readValueWithScalerAndUnit(eventData.get(3), scalerUnitDuration)));
+    } catch (final ObjectConfigException e) {
+      throw new ProtocolAdapterException("Error in object config", e);
```

Comment:
In general, we don't pay much attention in checking exceptions in the unit tests. I think we should allways conver them.

Reply:
Added exception checking with more specific messgages on deeper level methods. And added Unit test checking these exceptions accordingly.
The last exception catch in method getEventRelatedCosemSourceObject is actually checked in the cosem ObjectConfig and is not tested in unit test separately.